File: src/main/java/net/ccbluex/liquidbounce/utils/misc/SmoothStepAnimation.java
package net.ccbluex.liquidbounce.utils.misc;

public class SmoothStepAnimation extends Animation  {
    public SmoothStepAnimation(final int ms, final double endPoint) {
        super(ms, endPoint);
    }
    
    public SmoothStepAnimation(final int ms, final double endPoint, final Direction direction) {
        super(ms, endPoint, direction);
    }
    
    @Override
    protected double getEquation(final double x) {
        final double x2 = x / this.duration;
        return -2.0 * Math.pow(x2, 3.0) + 3.0 * Math.pow(x2, 2.0);
    }
}

File: src/main/java/net/ccbluex/liquidbounce/ui/client/clickgui/style/styles/nlclickgui/animations/impl/SmoothStepAnimation.kt
package net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.nlclickgui.animations.impl

import net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.nlclickgui.animations.Animation
import net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.nlclickgui.animations.Direction

class SmoothStepAnimation @JvmOverloads constructor(
    ms: Int,
    endPoint: Double,
    direction: Direction = Direction.FORWARDS
) : Animation(ms, endPoint, direction) {

    override fun getEquation(x: Double): Double {
        val x1 = x / duration.toDouble()
        return -2 * Math.pow(x1, 3.0) + 3 * Math.pow(x1, 2.0)
    }
}
